### PR TITLE
Create integration-Tanium.yml

### PR DESCRIPTION
**Pull request by shaniacht1**
fixes https://github.com/demisto/server/issues/6462
- [x] create docker image and change docker and pytan path in the integration accordingly
- [x] test playbook
- [x] delete Go integration
- [x] delete/update scripts using old integration

### CODE REVIEW

**Comment by shaniacht1** [comment]
### Overview:
Tanium is an endpoint management product that supports plain english questions.

### To set up [integration name] to work with Demisto:
Use the ip address, username and password used in the UI

### To set up the integration on Demisto:
For example:
1. Go to ‘Settings > Integrations > Servers & Services’
2. Locate [integration name] by searching for it using the search box on the top of the page.
3. Click ‘Add instance’ to create and configure a new integration. You should configure the following settings:  

   **Name**: A textual name for the integration instance.
   **Host**: The hostname or IP address of the appliance being used.
   **Appliance Port**: The appliance port being used.
   **Username and Password**: The username and password, or toggle to Credentials.
   **Test** plus troubleshooting steps if test fails

...

### Top 2-3 Use-cases:
The integration can be used to query the product in plain English. Use the ask-question command in order to do that. 
The integration can also be used in order to create a package and deploy it. User the package_args argument to pass argument to the command executed by the package. 

### Commands:

1. **Command Name**
tn-get-package
2. **Input**
name or id of package
3. **Context output (example)**
image

1. **Command Name**
tn-get-object
2. **Input**
name or id of object

1. **Command Name**
tn-get-saved-question
2. **Input**
name or id of saved question
3. **Context output (example)**
image

1. **Command Name**
!tn-get-all-packages
2. **Input**
no input
3. **Context output (example)**
same as get package

1. **Command Name**
!tn-get-all-saved-questions
2. **Input**
no input
3. **Context output (example)**
same as get question

1. **Command Name**
tn-create-package
2. **Input**
name, display_name, command etc.
3. **Context output (example)**
same as get package

1. **Command Name**
tn-deploy-package
2. **Input**
name of package to deploy, package_args
3. **Context output (example)**
none

1. **Command Name**
tn-ask-question
2. **Input**
any plain english query
3. **Context output (example)**
image

This integration was integrated and tested with build 6.5.314.4344 of Tanium and version 2.2.2 of pyTan"

**Comment by aviadl** [comment]
@shaniacht1 overall this documentation is a good start but please:
1. the context, get them in text not as image (print json?)
2. Please also add an example of raw output for each command.
3. last - i think that there are some commands that you didnt add context/output - if there is none, please mention

**Comment by Spivaka** [comment]
@aviadl don't we want the use case to describe which commands are used to run the use case?

**Comment by aviadl** [comment]
@shaniacht1 see @Spivaka comment, make sense to explain how to implement the use case

File: Integrations/integration-Tanium.yml [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+      description: Id of saved action to approve
+    outputs:
+    - contextPath: Tanium.ActionApproval.ApprovedFlag
+      description: Approval status
+    - contextPath: Tanium.ActionApproval.Id
+      description: Saved action id
+    - contextPath: Tanium.ActionApproval.Name
+      description: Saved action name
+    description: Approve saved actions
+  dockerimage: demisto/pytan
```
Comment (dorsha) [inline Integrations/integration-Tanium.yml]: Need to add to: requirements.txt

File: Scripts/script-TaniumShowPendingActions.yml [hunk 2]
```diff
@@ -148,3 +149,4 @@ dependson:
   must:
   - tn-get-object
 timeout: 0s
+releaseNotes: Deprecated script
```
Comment (dorsha) [inline Scripts/script-TaniumShowPendingActions.yml]: please write the command that replace this script.
Deprecated script, use `command name` instead.

File: Scripts/script-TaniumFindRunningProcesses.yml [hunk 3]
```diff
@@ -166,3 +167,4 @@ dependson:
   - tn-result-data
   - tn-result-info
 timeout: 0s
+releaseNotes: Deprecated script
```
Comment (dorsha) [inline Scripts/script-TaniumFindRunningProcesses.yml]: please write the command that replace this script.
Deprecated script, use `command name` instead.

File: Scripts/script-TaniumDeployAction.yml [hunk 4]
```diff
@@ -104,3 +105,4 @@ dependson:
   must:
   - tn-deploy-package
 timeout: 0s
+releaseNotes: Deprecated script
```
Comment (dorsha) [inline Scripts/script-TaniumDeployAction.yml]: please write the command that replace this script.
Deprecated script, use `command name` instead.

File: Scripts/script-TaniumAskQuestionComplex.yml [hunk 5]
```diff
@@ -165,3 +165,4 @@ dependson:
   - tn-result-info
   - tn-add-question-complex
 timeout: 0s
+releaseNotes: Deprecated script
```
Comment (dorsha) [inline Scripts/script-TaniumAskQuestionComplex.yml]: please write the command that replace this script.
Deprecated script, use `command name` instead.

File: Scripts/script-TaniumAskQuestion.yml [hunk 6]
```diff
@@ -128,3 +129,4 @@ dependson:
   - tn-result-data
   - tn-result-info
 timeout: 0s
+releaseNotes: Deprecated script
```
Comment (dorsha) [inline Scripts/script-TaniumAskQuestion.yml]: please write the command that replace this script.
Deprecated script, use `command name` instead.

File: Scripts/script-TaniumApprovePendingActions.yml [hunk 7]
```diff
@@ -157,5 +158,4 @@ dependson:
   - tn-add-object
   - tn-get-object
 timeout: 0s
-releaseNotes: "Added description"
-
+releaseNotes: Deprecated script
```
Comment (dorsha) [inline Scripts/script-TaniumApprovePendingActions.yml]: please write the command that replace this script.
Deprecated script, use `command name` instead.

File: Integrations/integration-Tanium.yml [hunk 8]
```diff
@@ -0,0 +1,483 @@
...
+        final_result = deployAction(handler)
+    if demisto.command() == 'tn-ask-system':
+        final_result = askQuestion(handler, 'Get Computer Name from all machines with Computer Name matching \"'+ demisto.args()['hostname'] + '\"')
+    if demisto.command() == 'tn-ask-question':
+        final_result = askQuestion(handler, demisto.args()['question'])
+    if demisto.command() == 'tn-create-package':
+        final_result = createPackage(handler)
+    if demisto.command() == 'tn-approve-saved-action':
+        final_result = approveSavedAction(handler)
+
```
Comment (dorsha) [inline Integrations/integration-Tanium.yml]: These command were supported in the server integration and now not.. it might break playbook for existing users.

tn-result-info
tn-result-data
tn-add-question-complex
tn-add-object

Also, need to make sure that arguments are still the same for all commands.

File: Integrations/integration-Tanium.yml [hunk 9]
```diff
@@ -0,0 +1,483 @@
...
+    restoreSoutAndExit(final_result)
+  type: python
+  commands:
+  - name: tn-get-package
+    arguments:
+    - name: name
```
Comment (dorsha) [inline Integrations/integration-Tanium.yml]: I think that one of name and id arguments should be required. and one of them should be default. No?

File: Integrations/integration-Tanium.yml [hunk 10]
```diff
@@ -0,0 +1,483 @@
...
+    - contextPath: Tanium.Packages.id
+      description: Tanium unique package id
+    - contextPath: Tanium.Packages.expire_seconds
+      description: Timeout for action expiry in seconds
+    - contextPath: Tanium.Packages.files
+      description: Packge files
+    description: Get all Tanium package objects
+  - name: tn-get-object
+    arguments:
+    - name: name
```
Comment (dorsha) [inline Integrations/integration-Tanium.yml]: I think that one of name and id arguments should be required. and one of them should be default. No?

File: Integrations/integration-Tanium.yml [hunk 11]
```diff
@@ -0,0 +1,483 @@
...
+      description: Question expire time in seconds
+    - contextPath: Tanium.SavedQuestions.id
+      description: Saved question unique id
+    - contextPath: Tanium.SavedQuestions.issue_seconds
+      description: Issue time in seconds
+    description: Gets all saved questions
+  - name: tn-deploy-package
+    arguments:
+    - name: package
+      required: true
```
Comment (dorsha) [inline Integrations/integration-Tanium.yml]: I would make it default argument

File: Integrations/integration-Tanium.yml [hunk 12]
```diff
@@ -0,0 +1,483 @@
...
+      description: Comma separated list of strings. Each string must describe a sensor
+        and a filter which limits which computers the action will deploy package to
+    - name: action_options
+      description: Comma separated list of options to apply to action_filters
+    - name: get_results
+      auto: PREDEFINED
+      predefined:
+      - "True"
+      - "False"
+      description: Specifies whether ait for result completion after deploying action
```
Comment (dorsha) [inline Integrations/integration-Tanium.yml]: ait should be wait?

File: Integrations/integration-Tanium.yml [hunk 13]
```diff
@@ -0,0 +1,483 @@
...
+    - name: package_args
+      description: Comma separated list of arguments neede to execute the package
```
Comment (dorsha) [inline Integrations/integration-Tanium.yml]: neede should be needed

File: Integrations/integration-Tanium.yml [hunk 14]
```diff
@@ -0,0 +1,483 @@
...
+        command
+    outputs:
+    - contextPath: Tanium.SavedActions.Id
+      description: Saved action id
+    - contextPath: Tanium.SavedActions.Name
+      description: Saved action name
+    description: Deploy a package and get the results back
+  - name: tn-ask-question
+    arguments:
+    - name: question
```
Comment (dorsha) [inline Integrations/integration-Tanium.yml]: make it default as well

File: Integrations/integration-Tanium.yml [hunk 15]
```diff
@@ -0,0 +1,483 @@
...
+      required: true
+      description: The question text
+    outputs:
+    - contextPath: Tanium.QuestionResults
+      description: Results of requested question. May be a complex object
+    description: Ask the server to parse the question text and choose the first parsed
+      result as the question to run
+  - name: tn-ask-system
+    arguments:
+    - name: hostname
```
Comment (dorsha) [inline Integrations/integration-Tanium.yml]: required and default?

File: Integrations/integration-Tanium.yml [hunk 16]
```diff
@@ -0,0 +1,483 @@
...
+      description: Name of host
+    outputs:
+    - contextPath: Tanium.QuestionResults
+      description: Results of requested computer name. May be a complex object
+    description: Ask a question about a specific endpoint
+  - name: tn-get-saved-question
+    arguments:
+    - name: name
```
Comment (dorsha) [inline Integrations/integration-Tanium.yml]: I think that one of name and id arguments should be required. and one of them should be default. No?

File: Integrations/integration-Tanium.yml [hunk 17]
```diff
@@ -0,0 +1,483 @@
...
+      description: Is the package deleted
+    - contextPath: Tanium.Packages.id
+      description: Tanium unique package id
+    - contextPath: Tanium.Packages.expire_seconds
+      description: Timeout for action expiry in seconds
+    description: Create a package object
+  - name: tn-approve-saved-action
+    arguments:
+    - name: action_id
+      required: true
```
Comment (dorsha) [inline Integrations/integration-Tanium.yml]: make default as well

**Comment by dorsha** [comment]
I would elaborate more about the "To set up Tanium to work with Demisto" part.

File: Integrations/integration-Tanium.yml (hunk 16 above)
Comment (shaniacht1) [reply Integrations/integration-Tanium.yml]: you need to use either one. if I set one to required you won't be able to use it without it.

File: Integrations/integration-Tanium.yml (hunk 10 above)
Comment (shaniacht1) [reply Integrations/integration-Tanium.yml]: you need to use either one. if I set one to required you won't be able to use it without it.

File: Integrations/integration-Tanium.yml (hunk 16 above)
Comment (dorsha) [reply Integrations/integration-Tanium.yml]: Ok, didn't know that

File: Integrations/integration-Tanium.yml (hunk 9 above)
Comment (shaniacht1) [reply Integrations/integration-Tanium.yml]: you need to use either one. if I set one to required you won't be able to use it without it.

File: Scripts/script-TaniumDeployAction.yml (hunk 4 above)
Comment (shaniacht1) [reply Scripts/script-TaniumDeployAction.yml]: The command is tn-deploy-package

File: Scripts/script-TaniumAskQuestionComplex.yml (hunk 5 above)
Comment (shaniacht1) [reply Scripts/script-TaniumAskQuestionComplex.yml]: No need for that one anymore, the new tn-ask-question supports plain english, so it already supports complex questions

File: Integrations/integration-Tanium.yml (hunk 8 above)
Comment (shaniacht1) [reply Integrations/integration-Tanium.yml]: info and data - were used to get results of other commands. But now this is already supported within the commands themselves (get_results arg)
tn-add-question-complex  -  ask-question now supports plain english, so there's no use for this one
tn-add-object - The command wasn't so useful without deep understanding of how the API works. I added the create-package command,

File: Integrations/integration-Tanium.yml (hunk 1 above)
Comment (shaniacht1) [reply Integrations/integration-Tanium.yml]: no pip install

**Comment by shaniacht1** [comment]
@mayairv I added to commands:
tn-get-all-pending-actions
tn-get-all-saved-actions

and tn-approve-saved-action was changed to tn-approve-pending-action

**Comment by dorsha** [comment]
@mayairv you didn't take the documentation from this issue.

**Comment by NeelRust** [comment]
@shaniacht1 are all then context output variables shown in below set by `tn-create-package`?

image
